iroha-cli/client.cpp

```diff
@@ -31,27 +31,38 @@
 #include "model/converters/pb_transaction_factory.hpp"
 #include "model/model_hash_provider_impl.hpp"
 
-namespace iroha_cli {
+#include <ed25519.h>
 
-  CliClient::CliClient(std::string target_ip, int port, std::string name)
-      : client_(target_ip, port) {
-    std::ifstream priv_file(name + ".priv");
-    priv_file >> client_priv_key_;
-    priv_file.close();
+namespace iroha_cli {
 
-    std::ifstream pub_file(name + ".pub");
-    pub_file >> client_pub_key_;
-    pub_file.close();
-  }
+  CliClient::CliClient(std::string target_ip, int port)
+      : client_(target_ip, port) {}
 
-  std::string CliClient::sendTx(std::string json_tx) {
+  CliClient::Status CliClient::sendTx(std::string json_tx) {
     iroha::ametsuchi::BlockSerializer serializer;
     auto tx_opt = serializer.deserialize(json_tx);
     if (not tx_opt.has_value()) {
-      return "Wrong transaction format";
+      return WRONG_FORMAT;
     }
-    iroha::model::converters::PbTransactionFactory factory;
     auto model_tx = tx_opt.value();
+    // Get private and public key of transaction creator
+    std::string client_pub_key_;
+    std::string client_priv_key_;
+    std::ifstream priv_file(model_tx.creator_account_id + ".priv");
+    if (not priv_file) {
+      return NO_KEYS;
+    }
+    priv_file >> client_priv_key_;
+    priv_file.close();
```

Comment:
No need to explicitly close stream - it is closed when the objects are destroyed.